### PR TITLE
Web Apps accessibility improvements - make apps and menus keyboard navigable

### PR DESCRIPTION
**Pull request by solleks**
## Summary
https://dimagi-dev.atlassian.net/browse/USH-394

This is version 2 of [an earlier change](https://github.com/dimagi/commcare-hq/pull/28902) that was reverted. It makes applications, menu items and case list items behave like links, generally using the following steps:

1. add the role="link" attribute which informs screen readers to treat it as a link.
2. put the element into the tab order with the tabindex="0" attribute, so it can receive keyboard focus.
3. add a keyboard event handler to the Marionette view for the element which treats Enter key presses like mouse clicks.
4. add a label with either aria-labelledby or aria-label to tell a screen reader what text to read for the element.

The flaw in version 1 was that the CaseView Marionette view had an ARIA label and role that prevented users from hearing the individual table data elements with a screen reader, so they no longer have those attributes.

This version also makes the delete button (trashcan) for incomplete sessions work with the keyboard and it avoids any changes to pagination controls to prevent a conflict with another concurrent change.

## Product Description
For users without a screen reader, this change enables keyboard navigation of Web Apps, using the Tab and Enter keys. Navigating by keyboard is one way to test that it works. No other changes should be evident without a screen reader.

## Safety Assurance

- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### QA Plan

[QA ticket](https://dimagi-dev.atlassian.net/browse/QA-2791)

This change should also be reviewed in staging by our screen reader expert who flagged the issue with the previous version of the change.

### Safety story
WAI-ARIA attributes are designed to have no effect outside of accessibility tools, so they shouldn't cause any regressions.

The other changes are putting elements into the focus ring and handling key clicks on them, which should not affect existing users who navigate by mouse clicks.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Comment by solleks** [comment]
Hi @stephherbers. Let's see if we can get this change pushed to staging, reviewed and QA tested.

**Comment by solleks** [comment]
QA has completed successfully ([ticket](https://dimagi-dev.atlassian.net/browse/QA-2791)).

**Comment by solleks** [comment]
This PR passed its evaluation by our screen reader expert.

**Comment by solleks** [comment]
Hi @orangejenny. If you have time to look at this change, it has passed both Dimagi QA and a check by our screen reader expert.

File: corehq/apps/cloudcare/templates/formplayer/grid_view.html [hunk 1]
```diff
@@ -111,7 +111,7 @@ <h3>{% trans "Settings" %}</h3>
   <div class="container">
     <div class="spacer"></div>
     <div class="col-xs-6 col-sm-4 col-xs-offset-3 col-sm-offset-4">
-      <div class="appicon appicon-default js-start-app">
+      <div class="appicon appicon-default js-start-app" role="link" tabindex="0" aria-label="<%= appName %>">
```
Comment (orangejenny) [inline corehq/apps/cloudcare/templates/formplayer/grid_view.html:114]: The app name should use `<%- ... %>` so it gets escaped. We just did a bunch of escaping fixes in https://github.com/dimagi/commcare-hq/pull/29471

**Comment by orangejenny** [comment]
It's great to hear testing went well. @stephherbers can you handle review & merge of this?

File: corehq/apps/cloudcare/templates/formplayer/grid_view.html (hunk 1 above)
Comment (solleks) [reply corehq/apps/cloudcare/templates/formplayer/grid_view.html]: OK, done.

**Review by stephherbers — APPROVED** [review]

**Comment by stephherbers** [comment]
Thank you for your work on this @solleks-- everything looks great

**Pull request merged**